homeassistant/components/tankerkoenig/entity.py

```diff
@@ -23,7 +23,7 @@ def __init__(
         super().__init__(coordinator)
         self._attr_device_info = DeviceInfo(
             identifiers={(ATTR_ID, station.id)},
-            name=f"{station.brand} {station.street} {station.house_number}",
+            name=f"{station.brand} {'' if station.street is None else station.street} {'' if station.house_number is None else station.house_number}",
```

Comment:
In case both are empty,  this would add some empty spaces in between,  which would not be expected,  if someone uses this value in an automation

Reply:
Yes, agreed, suggestions? Is there some kind of `join` that can handle empty elements?